### PR TITLE
Spin command to allow spinning of any rings any direction any amount

### PR DESCRIPTION
**Pull request by Solaxi**
Changed /custom spin_tiles_in_first_three_rings command to allow custom rotations. 

* Renamed to /custom spin_tiles_in_rings
* Kept the original behaviour as default.
* Can also give optional flavour message to be sent to game main channel. 

Example:
image

### CODE REVIEW

File: src/main/java/ti4/commands/custom/SpinTilesInRings.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+      List<Tile> tilesToSet = new ArrayList<>();
+      for (String spinString : customSpins) {
+
+        String[] spinSettings = spinString.split(":");
+        if (spinSettings.length != 3) {
+          MessageHelper.sendMessageToChannel(event.getMessageChannel(), "Invalid spin settings: " + spinString);
+          return;
+        }
+
+        int ring = Integer.parseInt(spinSettings[0]);
```
Comment (jamesrudolph) [inline src/main/java/ti4/commands/custom/SpinTilesInRings.java:47]: I would throw these parseInts into a try/catch or some other validation that they are ints. Maybe just a regex pattern check to handle all validation

File: src/main/java/ti4/commands/custom/SpinTilesInRings.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        Game activeGame = getActiveGame();
+
+        if (event.getOption(Constants.CUSTOM) == null) {
+          spinRings(activeGame);
+        } else {
+          spinRingsCustom(activeGame, event);
+        }
+    }
+
+    public void spinRingsCustom(Game activeGame, SlashCommandInteractionEvent event) {
```
Comment (jamesrudolph) [inline src/main/java/ti4/commands/custom/SpinTilesInRings.java:36]: add static

File: src/main/java/ti4/commands/custom/SpinTilesInRings.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+      activeGame.rebuildTilePositionAutoCompleteList();
+      OptionMapping flavourMsg = event.getOption(Constants.MESSAGE);
+      MessageHelper.sendMessageToChannel(activeGame.getMainGameChannel(), flavourMsg != null ? flavourMsg.getAsString() : "Spun the rings");
+    }
+
+    //the original spin logic which does
+    // - ring 1 cw one step
+    // - ring 2 ccw two steps
+    // - ring 3 cw three steps (except 6p map HS positions)
+    public void spinRings(Game activeGame){
```
Comment (jamesrudolph) [inline src/main/java/ti4/commands/custom/SpinTilesInRings.java:88]: add static

File: src/main/java/ti4/commands/status/Cleanup.java [hunk 4]
```diff
@@ -101,7 +101,7 @@ public void runStatusCleanup(Game activeGame) {
         activeGame.setStoredValue("PublicExecution", "");
         activeGame.setHasHadAStatusPhase(true);
         if(activeGame.isSpinMode()){
-            new SpinTilesInFirstThreeRings().spinRings(activeGame);
```
Comment (jamesrudolph) [inline src/main/java/ti4/commands/status/Cleanup.java:104]: remove new

**Review by jamesrudolph — APPROVED** [review]

**Comment by Solaxi** [comment]
Implemented suggested improvements!

**Pull request merged**